### ISSUE TITLE
No notifications when I safely eject USB drives

### ISSUE THREAD

**Issue by glu8716** [lxqt/lxqt#2316]
##### Expected Behavior

When I eject a removable device through the "Removable Media" panel widget or through PCManFM I should get a notification saying it is now safe to eject it.

##### Current Behavior

I don't get any notification.

##### Steps to Reproduce (for bugs)

1. Plug a removable drive, such as a USB stick
2. Mount it
3. Eject it through PCManFM or through the "Removable Media" panel widget

##### Context

No particular context. I just remember I was receiving notifications some weeks ago. Not they suddenly doesn't show up anymore.

##### System Information

* Distribution & Version: Artix
* Kernel: 5.19.13-zen
* Qt Version: 5.15.6
* liblxqt Version: 1.1.0
* Package version: 3

**Comment by tsujan** [comment]
Not reproducible with the latest git.

**Comment by stefonarch** [comment]
On my side it's several months I don't see any notification anymore, with latest git on my laptop. Have to check the debian PC with LXQt 1.1 where I think I still see those notification about successful removed devices.

I never saw any settings about omitting those notification.

**Comment by stefonarch** [comment]
Checked on debiab LXQt 1.1.0 and also there I see no notification anymore.

**Comment by glu8716** [comment]
I looked everywhere and apparently the culprit is a checkbox in the "Removable Media" widget options. There's a "When a device is connected" > "Show info" option there. Looks like that notifications are sent only if that option is set to "Show info".
Since there's also a "When a eject button is pressed" option, which doesn't have a "Show info" option, I haven't thought the problem could be there. I think this is a little bit unintuitive and should be redesigned if it's the root of the "problem".
I also think the possibility to see notifications about removable devices shouldn't be tied to a panel widget, but should be a global option.

**Comment by stefonarch** [comment]
Yep, confirmed, I played around with all combinations of options in pcmanfm-qt and some in the panel plugin but that was the only one which I didn't check. 

Let's here others, but maybe only the text needs to be changed to "Notify about connect/remove" or similar.

**Comment by tsujan** [comment]
I don't get what the problem is. There are three options, whose texts seem self-explanatory: Popup menu, Show info, Do nothing. It has always been so, while the report pretends that something has changed (hence closing).

**Comment by stefonarch** [comment]
I happened to have the same experience as @glu8716: notifications stopped working, I didn't bother about.

IMHO "Show info" is  not clear about notifications. "Show notifications and popup-menu" is what is does.

**Comment by tsujan** [comment]
> notifications stopped working

That's what I don't understand. If notifications stopped working, something should have been changed, while you say that the option in question works.

**Comment by glu8716** [comment]
The option in question works indeed. But we _thought_ it wasn't working because it isn't clear (at least to me, since I'm relatively new to LXQt) what makes the notification appears or not. The reason I opened this issue is that I didn't know devices notifications were tied to the panel's widget, which I had removed at a later time. And if I read "Show info when a device is connected" I assume it's some kind of info that appears when a device _is connected_, and not disconnected.

The fact that it has always been so doesn't mean it can't be better. Maybe something like a checkbox with "Show devices notifications" or something like that could be clearer. And I think it should be something that doesn't depend on a panel widget. What if, for example, I want notifications but don't use the LXQt panel?

**Comment by tsujan** [comment]
Quoiting: "I just remember I was receiving notifications some weeks ago...."

**Comment by glu8716** [comment]
> The reason I opened this issue is that I didn't know devices notifications were tied to the panel's widget, which I had removed at a later time.

Initially I had the widget on the panel with that setting I unconsciously set to "show info", therefore I was receiving notifications. At that time I didn't think notifications were tied to the widget, I thought showing notifications was a normal behavior for LXQt.
When at a later time I removed the widget from the panel and didn't receive notifications anymore I opened this issue. Only after opening it I realized the widget, if not present on the panel, is the reason for not having notifications. 

So, yeah, this is not really a "issue" anymore, but rather a suggestion now, as I think this setting could be clearer :)

**Comment by tsujan** [comment]
OK.

I have no problem with changing of "Show info", although it seems clear enough to me (and to many other users)..

Showing notifications for removable media needs a program running in the background. A panel plugin called "Removable Media" seems like an obvious choice.

I change the title because it was very misleading (I was going to check the changelog of `solid` when I first saw it).

**Comment by stefonarch** [comment]
For me notifications stopped when I changed months ago this setting by case.
It looks also exclusive: Show info ><  Popup menu while the popup menu is shown also with "Show info".

I see  2 options to improve  clearness:
* Change text to something "Popup menu and notifications"
* Replace this option with a checkbox "Show device  notifications"

**Comment by glu8716** [comment]
Yeah. I'm sorry for the misunderstanding.

To be honest to me the problem isn't really the "Show info" text (although I admit that changing it to "notifications" seems to be more straightforward), but the fact that the notifications when ejecting are set by a field called "When a device is **connected**" instead of the "When **eject button is pressed**" one. In this case my suggestion would be to add the same "Show info" (or whatever) to the "When eject button is pressed" field too for more consistency. Or making a global "Show notifications" checkbox.

**Comment by stefonarch** [comment]
That's a good point. So a checkbox "Notify about devices"?

On a  side note I just saw that whith  a drive containing 2 partitions (one is encrypted) only the "connected" notifications do appear, while no info at all on unmount is shown. Probably related to https://github.com/lxqt/lxqt-panel/issues/1639  , I have no drive with 2 normal partitions to test atm.

**Comment by tsujan** [comment]
This report might not be seen here. Transferring it to lxqt-panel's issue tracker....
